### PR TITLE
handle loading dataset error

### PR DESCRIPTION
**Pull request by chinmaygupta28**
When -LOADING dataset into memory error comes from the redis, we can
safely retry those requests. We are failing that server at the moment
it receives the loading error, there is another thread which runs and
make the PING PONG request every refresh interval (default is 10s I
think). PONG will be successful and it will make the server as not
failing and then again it will retried.

### CODE REVIEW

File: src/Handler.cpp [hunk 1]
```diff
@@ -802,6 +802,15 @@ void Handler::handleResponse(ConnectConnection* s, Request* req, Response* res)
                 c->peer(), c->fd(), c->status(), c->statusStr());
         return;
     }
+    if (res->type() == Reply::Error) {
+        if (res->isLoading()) {
+            Server* serv = s->server();
+            serv->setFail(true);
+            logNotice("server %s mark failure because it is loading error", serv->addr().data());
+            handleRequest(req);
```
Comment (gunjanpatidar) [inline src/Handler.cpp]: can potentially go in infinite loop. Should there be an upper cap on retries?

Comment (chinmaygupta28) [reply src/Handler.cpp]: Yes, will add this.

Comment (chinmaygupta28) [reply src/Handler.cpp]: Found another bug, will fix both in the next commit.

Comment (chinmaygupta28) [reply src/Handler.cpp]: fixed in a11d93f, testing this more with integration and unit

**Pull request closed without merging**